runtime/runtime_impl_jetty12/src/main/java/com/google/apphosting/runtime/jetty/JettyServletEngineAdapter.java

```diff
@@ -118,27 +120,45 @@ public InvocationType getInvocationType() {
             return InvocationType.BLOCKING;
           }
         };
-    rpcConnector =
-        new DelegateConnector(server, "RPC") {
-          @Override
-          public void run(Runnable runnable) {
-            // Override this so that it does the initial run in the same thread.
-            // Currently, we block until completion in serviceRequest() so no point starting new
-            // thread.
-            runnable.run();
-          }
-        };
-    server.addConnector(rpcConnector);
+
+    // Don't add the RPC Connector if in HttpConnector mode.
+    if (!isHttpConnectorMode)
+    {
+      rpcConnector =
```

Comment:
I understand that the rpcConnector won't be needed for HTTP mode but with this change we are no longer initializating this field anymore for both modes. Any chance that it would cause an [NPE below](https://github.com/GoogleCloudPlatform/appengine-java-standard/pull/309/files#diff-c91b2a5ea1df3ab41597d963e4de404be5ea3ce126d041a6dd7c15da42686eb3R265) for HttpConnector when we actually service the request?

Reply:
This field is only ever used in `JettyServletEngineAdaptor.serviceRequest`, and in the GEN2 runtimes the only way we get to `serviceRequest` is through the `ForwardingHandler` in the `JettyHttpProxy`, and in HttpConnectorMode we never add this handler to go over RPC.